src/data/prisma_client.py

```
@@ -62,16 +62,39 @@ def execute_prisma_command(script: str) -> Dict[str, Any]:
+            # Validate script file exists and is readable
+            script_path = "temp_prisma_script.js"
+            if not os.path.exists(script_path):
+                raise Exception(f"Script file {script_path} does not exist")
+
+            # Security: Ensure script file is not writable by others
+            script_stat = os.stat(script_path)
+            if script_stat.st_mode & 0o022:  # Check if group or others can write
+                raise Exception(f"Script file {script_path} has insecure permissions")
+
+            # Security: Node path validated with shutil.which(), script path validated for existence and permissions
             result = subprocess.run(
```

Ruff (0.13.0)

`subprocess` call: check for execution of untrusted input

(S603)